### PR TITLE
Update dependencies in E2E tests

### PR DESCRIPTION
**Pull request by heng-liu**
## Bug

Fixes: https://github.com/NuGet/Client.Engineering/issues/1818

Regression? Last working version:

## Description
Update the dependency versions in E2E tests.

## PR Checklist

- [ ] PR has a meaningful title
- [ ] PR has a linked issue.
- [ ] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [ ] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [ ] N/A

### CODE REVIEW

**Review by dominoFire — DISMISSED** [review]

**Review by nkolev92 — DISMISSED** [review]

File: test/EndToEnd/ProjectTemplates/UwpPackageRefClassLibrary.zip/ClassLibrary47.csproj [hunk 1]
```diff
@@ -11,8 +11,8 @@
     <AssemblyName>$safeprojectname$</AssemblyName>
     <DefaultLanguage>en-US</DefaultLanguage>
     <TargetPlatformIdentifier>UAP</TargetPlatformIdentifier>
-    <TargetPlatformVersion Condition=" '$(TargetPlatformVersion)' == '' ">10.0.18362.0</TargetPlatformVersion>
-    <TargetPlatformMinVersion>10.0.10586.0</TargetPlatformMinVersion>
+    <TargetPlatformVersion Condition=" '$(TargetPlatformVersion)' == '' ">10.0.19041.0</TargetPlatformVersion>
+    <TargetPlatformMinVersion>10.0.17763.0</TargetPlatformMinVersion>
```
Comment (zivkan) [inline test/EndToEnd/ProjectTemplates/UwpPackageRefClassLibrary.zip/ClassLibrary47.csproj]: I didn't look into the CI failures, but be careful about changing these values. One of the two values determines which version of the Windows SDK VS will look for on the machine, and if that version of the Windows SDK isn't defined in the vsconfig file that the DartLab machine uses for configuration, VS will pop up a dialog that will freeze the test until the test, or the entire pipeline, times out.

Comment (heng-liu) [reply test/EndToEnd/ProjectTemplates/UwpPackageRefClassLibrary.zip/ClassLibrary47.csproj]: Good catch! Thanks! Fixed.

**Review by jeffkl — APPROVED** [review]

**Review by dominoFire — APPROVED** [review]
Maybe we can use `dotnet new` to create new test projects on the fly.

File: test/EndToEnd/tests/BuildIntegratedTest.ps1 [hunk 2]
```diff
@@ -216,7 +216,7 @@ function Test-BuildIntegratedProjectClosureWithLegacyProjects {
 function Test-BuildIntegratedMixedLegacyProjects {
     # Arrange
     $project1 = New-ClassLibrary
-    $project1 | Install-Package Newtonsoft.Json -Version 5.0.6
+    $project1 | Install-Package Newtonsoft.Json -Version 13.0.1
```
Comment (jeffkl) [inline test/EndToEnd/tests/BuildIntegratedTest.ps1:219]: Is it easy to declare a version variable so this can be updated in one place next time if needed?

**Pull request merged**